packages/components/addon/components/hds/tabs/tab.hbs

```diff
@@ -1,6 +1,9 @@
 {{! template-lint-disable require-context-role }}
 {{! template-lint-disable no-invalid-role }}
 
+{{! Temporary: fix linter errors with using FlightIcon and BadgeCount inside role="tab" element }}
+{{! template-lint-disable require-presentational-children }}
```

Comment:
The linter was complaining after I added these. Any suggestions on how to restructure the html DOM? @MelSumner

Reply:
**Current DOM Structure (simplified):**
```
<li role="presentation">
  <a role="tab" href="#blah">
    <FlightIcon />
    <span>{{yield}}</span>
    <BadgeCount />
  </a>
</li>
```

**I could do something like below instead to satisfy the linter:**
(I could maybe get rid of the inner wrapping span tag.)
```
<li role="presentation">
  <span>
    <FlightIcon />
    <a role="tab" href="#blah">{{yield}}</a>
    <BadgeCount />
  </span>
</li>
```

I could use CSS to expand the target area of the nested anchor tab to cover the full tab area. @MelSumner What do you think or do you have alternate suggestions?

Reply:
@KristinLBradley I was working on an experiment for the layout, a while back- want to check it out and see what you think? [CodePen: [a11y] tabs: accessible overflow](https://codepen.io/melsumner/pen/PoRgbVv??editors=1000).

We can make time to pair on this today if you want, maybe some 🦆 will help. 👍

Reply:
Thanks. To me it makes sense to use links but I can use buttons instead if that's what you prefer. I tried using a button with the icons inside though and the linter still complained so it doesn't solve that issue. So I'd still have to restructure the DOM to move the icons outside of the button. Unless it's ok to disable the linter for this?

`{{! template-lint-disable require-presentational-children }}`

Reply:
I'm going to go ahead and resolve this since I opened it but will continue to work on it in the next iteration. (I'd like to get the other updates I made merged in.)

Reply:
I unresolved again. Sorry for any confusion. So anyway I can go ahead and refactor the DOM so I can delete the lint disable. Would really like to know if you want me to use buttons instead of links though. (As I said my preference is to use links but if others want buttons instead I can change it.)

Reply:
Can you explain why you'd rather use a link than a button? I'm open to the conversation, I just think a button element is the more appropriate element in this case.

Even if a query param is being appended, we want to still use the button, I think. Using a link will cause the router to view the whole thing as a new URL and there are implications for this in accessible routing in Ember apps.

Reply:
It made sense to me semantically but I'm not trying to push one over the other. I started out using links in the initial basic HTML implementation and didn't receive any feedback to use a button instead. I'll change it to use a button.

Reply:
I refactored the DOM to fix the linter error and to use a button instead of a link.

Reply:
Ok I'll fix it, thanks